rpm/prebuild.sh

```diff
@@ -0,0 +1,37 @@
+set_python3() {
+    python3_default=/usr/bin/python3
+    for v in `seq 9 -1 4`
+    do
+      python3=/usr/bin/python3.$v
+      if [[ -e $python ]] && [[ ! -e $python3_default ]]; then
+        echo "Create symlink for Python 3: $python3_default --> $python3"
+        sudo ln -s $python3_default $python3
+        break
+      fi
+    done
+}
+
+if type yum; then
+    VER=$(rpm -E %{rhel})
+    # No testing on CentOS 6
+    [[ $VER == 6 ]] && exit 0
+    sudo yum install -y python3
+    set_python3
+    if [[ $VER == 7 ]]; then
+        sudo yum install -y python3-devel gcc
+        python3 -m pip install --user -r test-run/requirements.txt
+    fi
+elif type dnf; then
+    VER=$(rpm -E %{rhel})
+    # No testing on CentOS 6
+    [[ $VER == 6 ]] && exit 0
+    sudo dnf install -y python3
+    set_python3
+    if [[ $VER == 7 ]]; then
+        sudo dnf install -y python3-devel gcc
+        python3 -m pip install --user -r test-run/requirements.txt
+    fi
+else
+    echo "No suitable package manager found"
+    exit 1
+fi
```

Comment:
There are `python3-six`, `python3-gevent` and `python3-pyyaml` packages on all RPM-based distros we support except CentOS 7. So there is no any reason to setup weird hacks with mangling a build system and installing implicit dependencies: we can just rebuild three packages for CentOS 7. That's all. All modern distros already have everything that we need and so the future versions of those distros will have it too. Zero effort.

I already uploaded those three packages to [packpack/backports](https://packagecloud.io/packpack/backports)[^1]. This repository is already enabled in the `packpack/packpack:el-7` docker image.

In fact, the whole problem appears from your wish to use non-default Python interpreter on CentOS 7. But okay. Let's remove the prebuild script and setup correct dependencies in the RPM spec. Python 2 ones or Python 3 ones — does not matter.

<details>
<summary>The list of relevant packages on all distros we support (or will support soon)</summary>

centos:7

[root@04084cd6fdb5 /]# yum search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
<..nothing..>

centos:7 + epel

[root@a89d9922e638 /]# yum search python3 | grep -i '^python3.\?-\(gevent\|six\|yaml\|pyyaml\)\.'
python34-six.noarch : Python 2 and 3 compatibility utilities
python36-PyYAML.x86_64 : YAML parser and emitter for Python 3.6
python36-six.noarch : Python 2 and 3 compatibility utilities

centos:8

[root@805838c967a1 /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Failed to set locale, defaulting to C.UTF-8
Last metadata expiration check: 0:00:01 ago on Fri Feb 12 18:05:07 2021.
python3-gevent.x86_64 : A coroutine-based Python networking library
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-six.noarch : Python 2 and 3 compatibility utilities

fedora:28

[root@2fb0a0b21a3a /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:00:37 ago on Fri Feb 12 18:09:59 2021.
python3-six.noarch : Python 2 and 3 compatibility utilities
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-PyYAML.x86_64 : YAML parser and emitter for Python
python3-gevent.x86_64 : A coroutine-based Python networking library

fedora:29

[root@ae3e60e3eed0 /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:00:32 ago on Fri 12 Feb 2021 06:12:44 PM UTC.
python3-six.noarch : Python 2 and 3 compatibility utilities
python3-six.noarch : Python 2 and 3 compatibility utilities
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-gevent.x86_64 : A coroutine-based Python networking library

fedora:30

[root@d4b31068f50e /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:11:14 ago on Fri Feb 12 18:14:16 2021.
python3-six.noarch : Python 2 and 3 compatibility utilities
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-gevent.x86_64 : A coroutine-based Python networking library

fedora:31

[root@99fe0bf111ff /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:01:00 ago on Fri Feb 12 18:26:28 2021.
python3-six.noarch : Python 2 and 3 compatibility utilities
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-gevent.x86_64 : A coroutine-based Python networking library

fedora:32

[root@213025c8eabf /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:03:36 ago on Fri Feb 12 18:28:27 2021.
python3-gevent.x86_64 : A coroutine-based Python networking library
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-six.noarch : Python 2 and 3 compatibility utilities

fedora:33

[root@392f0990a852 /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:06:13 ago on Fri Feb 12 18:33:11 2021.
python3-gevent.x86_64 : A coroutine-based Python networking library
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-six.noarch : Python 2 and 3 compatibility utilities

fedora:34

[root@4c0f498f17b3 /]# dnf search python3- | grep -i '^python3-\(gevent\|six\|yaml\|pyyaml\)\.'
Last metadata expiration check: 0:03:12 ago on Fri Feb 12 18:00:58 2021.
python3-gevent.x86_64 : A coroutine-based Python networking library
python3-pyyaml.x86_64 : YAML parser and emitter for Python
python3-six.noarch : Python 2 and 3 compatibility utilities

opensuse/leap:15.0

a3fe2bf476c3:/ # zypper search python3- | grep -i ' python3-\(gevent\|six\|yaml\|pyyaml\) '
  | python3-PyYAML                                 | YAML parser and emitter for Python                                                 | package
  | python3-gevent                                 | Python network library that uses greenlet and libevent                             | package
  | python3-six                                    | Python 2 and 3 compatibility utilities                                             | package

opensuse/leap:15.1

79d985d6745a:/ # zypper search python3- | grep -i ' python3-\(gevent\|six\|yaml\|pyyaml\) '
  | python3-PyYAML                                          | YAML parser and emitter for Python                                                    | package
  | python3-gevent                                          | Python network library that uses greenlet and libevent                                | package
  | python3-six                                             | Python 2 and 3 compatibility utilities                                                | package

opensuse/leap:15.2

7297a2fecf02:/ # zypper search python3- | grep -i ' python3-\(gevent\|six\|yaml\|pyyaml\) '
  | python3-PyYAML                                          | YAML parser and emitter for Python                                                    | package
  | python3-gevent                                          | Python network library that uses greenlet and libevent                                | package
  | python3-six                                             | Python 2 and 3 compatibility utilities                                                | package

opensuse/leap:15.3

29c77b1b09f6:/ # zypper search python3- | grep -i ' python3-\(gevent\|six\|yaml\|pyyaml\) '
  | python3-PyYAML                                          | YAML parser and emitter for Python                                                    | package
  | python3-gevent                                          | Python network library that uses greenlet and libevent                                | package
  | python3-six                                             | Python 2 and 3 compatibility utilities                                                | package
</details>

[^1]: The source packages are [from CentOS 8](https://vault.centos.org/8.3.2011/AppStream/Source/SPackages/).

Reply:
> I already uploaded those three packages to packpack/backports[^1]. This repository is already enabled in the packpack/packpack:el-7 docker image.

Thanks for this!

Removed line with `pip` and enhanced a list of packages for case when `$VER == 7`.

Reply:
script `rpm/prebuild.sh` has been removed

Reply:
So, finally, our packages building works with only packaged dependencies? That is nice to hear!

Reply:
> So, finally, our packages building works with only packaged dependencies?

Yep.